From 83aaff839cf3b6788d95a253bb099992b366136d Mon Sep 17 00:00:00 2001
From: Kent Fredric <kentnl@gentoo.org>
Date: Fri, 16 Mar 2018 21:09:54 +1300
Subject: Disable auto-probes for Date-Manip

These are currently fragile and broken, and ultimately unnecessary
on Gentoo (because we have the luxury of forcing a TZ, and if
Date::Manip still doesn't work, that's a bigger bug we care about )

Bug: https://rt.cpan.org/Public/Bug/Display.html?id=85001
---
 t/doc.t   | 15 ---------------
 t/funcs.t | 14 --------------
 t/manip.t | 17 -----------------
 3 files changed, 46 deletions(-)

diff --git a/t/doc.t b/t/doc.t
index c5bfe6f..b9ce2af 100755
--- a/t/doc.t
+++ b/t/doc.t
@@ -9,19 +9,6 @@ BEGIN { $Time::Format::NOXS = 1 }
 BEGIN { use_ok 'Time::Format', qw(:all) }
 my $tl_notok;
 BEGIN { $tl_notok = eval('use Time::Local; 1')? 0 : 1 }
-my $dm_notok;
-my $dm_notz;
-BEGIN
-{
-    $dm_notok = eval('use Date::Manip (); 1')? 0 : 1;
-    unless ($dm_notok)
-    {
-        # If Date::Manip can't determine the time zone, it'll bomb out of the tests.
-        $dm_notz = eval('Date::Manip::Date_TimeZone (); 1')? 0 : 1;
-    }
-    delete $INC{'Date/Manip.pm'};
-    %Date::Manip:: = ();
-}
 
 # Were all variables imported? (3)
 is ref tied %time,     'Time::Format'   =>  '%time imported';
@@ -83,8 +70,6 @@ SKIP:
     # manip tests (3)
     SKIP:
     {
-        skip 'Date::Manip not available',             3  if $dm_notok;
-        skip 'Date::Manip cannot determine timezone', 3  if $dm_notz;
         is $manip{'%m/%d/%Y',"epoch $t"},                       '06/05/2003'    => 'Example 13';
         is $manip{'%m/%d/%Y','first monday in November 2000'},  '11/06/2000'    => 'Example 14';
         is qq[$time{'yyyymmdd',$manip{'%s',"epoch $t"}}],       '20030605'      => 'Example 15';
diff --git a/t/funcs.t b/t/funcs.t
index 1e5bdb6..88f3fdc 100755
--- a/t/funcs.t
+++ b/t/funcs.t
@@ -13,18 +13,6 @@ BEGIN {
     delete $INC{'POSIX.pm'};
     %POSIX:: = ();
 }
-my $manip_bad;
-my $manip_notz;
-BEGIN {
-    $manip_bad = eval('use Date::Manip (); 1')? 0 : 1;
-    unless ($manip_bad)
-    {
-        # If Date::Manip can't determine the time zone, it'll bomb out of the tests.
-        $manip_notz = eval ('Date::Manip::Date_TimeZone (); 1')? 0 : 1;
-    }
-    delete $INC{'Date/Manip.pm'};
-    %Date::Manip:: = ();
-}
 
 # Get day/month names in current locale
 my ($Thursday, $Thu, $June, $Jun);
@@ -71,8 +59,6 @@ SKIP:
     # time_manip tests (6)
     SKIP:
     {
-        skip 'Date::Manip not available',             6 if $manip_bad;
-        skip 'Date::Manip cannot determine timezone', 6 if $manip_notz;
         my $m = 'first thursday in june 2003';
         is time_manip('%Y',$m),      '2003'      => 'year';
         is time_manip('%d',$m),      '05'        => 'day of month';
diff --git a/t/manip.t b/t/manip.t
index 763ad59..6f94607 100755
--- a/t/manip.t
+++ b/t/manip.t
@@ -5,28 +5,11 @@ use Test::More tests => 6;
 
 BEGIN { $Time::Format::NOXS = 1 }
 BEGIN { use_ok 'Time::Format', qw(%manip) }
-my $manip_bad;
-BEGIN
-{
-    unless (eval 'use Date::Manip (); 1')
-    {
-        $manip_bad = 'Date::Manip is not available';
-    }
-    else
-    {
-        # If Date::Manip can't determine the time zone, it'll bomb out of the tests.
-        $manip_bad = 'Date::Manip cannot determine time zone'
-            unless eval 'Date::Manip::Date_TimeZone(); 1';
-    }
-    delete $INC{'Date/Manip.pm'};
-    %Date::Manip:: = ();
-}
 
 my $t = 'first thursday in june 2003';
 
 SKIP:
 {
-    skip $manip_bad, 5 if $manip_bad;
     is $manip{'%Y',$t},      '2003'      => 'year';
     is $manip{'%d',$t},      '05'        => 'day of month';
     is $manip{'%D',$t},      '06/05/03'  => '%D';
-- 
2.16.2

